### DISCUSSION TITLE
problem to train Padim model

### DISCUSSION THREAD

**Discussion by nargolakbari** [open-edge-platform/anomalib#838] [General]
Hello everyone.
I have a problem to train Padim model , I use Colab version and start running all commands without any changes , but when I want to train model with > **!python tools/train.py --config anomalib/models/padim/config.yaml** I got this error :

Traceback (most recent call last):
  File "tools/train.py", line 19, in <module>
    from anomalib.data.utils import TestSplitMode
ImportError: cannot import name 'TestSplitMode' from 'anomalib.data.utils' (/usr/local/lib/python3.8/dist-packages/anomalib/data/utils/__init__.py)

it seems that **train.py** does not consider anomalib in this path  **/content/anomalib** and try to find modules in wrong path . how can force it to use right path ?
I would be grateful if you help me .

a part of **train.py** ;
,,,

from anomalib.config import get_configurable_parameters
from anomalib.data import get_datamodule
from anomalib.data.utils import TestSplitMode
from anomalib.models import get_model
from anomalib.utils.callbacks import LoadModelCallback, get_callbacks
from anomalib.utils.loggers import configure_logger, get_experiment_logger
.....

**Comment by samet-akcay** [comment]
@nargolakbari, `TestSplitMode` has recently been added to the library. Can you check if you pull the latest version of the library and install it ?

**Reply by nargolakbari** [reply]
@samet-akcay Thanks for your response
 I solve this problem with adding this lines :
import sys
sys.path.append('/content/drive/MyDrive/anomalib')
because it can not find right one **but now there is another problem while downloading datasets/MVTec :**
...
   download_and_extract(self.root, DOWNLOAD_INFO)
  File "/content/drive/MyDrive/model1/anomalib-main/anomalib/data/utils/download.py", line 248, in download_and_extract
    raise ValueError(f"Unrecognized file format: {downloaded_file_path}")
ValueError: Unrecognized file format: datasets/MVTec/420938113-1629952094

it seems that the downloaded file is unknown, please help me  because I need to run this code for my project . 
Thank you again

**Reply by samet-akcay** [reply]
this looks like the same issue as reported here #835. We'll have a look at the issue today and let you know.

**Reply by nargolakbari** [reply]
Thank you so much .

**Reply by djdameln** [reply]
Hi @nargolakbari, the error which you reported was caused by some recent changes to the dataset downloading functionality. We just merged [a fix](https://github.com/openvinotoolkit/anomalib/pull/842) for this bug. If you pull the latest changes to main branch, the MVTec dataset should download and extract without any issues. Please let us know if you still encounter any problems!

**Reply by nargolakbari** [reply]
@djdameln thanks  , I tested the code but error is still there , if you test code with Colab you can see it .

**Reply by nargolakbari** [reply]
and the first problem is also still there when you run Colab version , for this command: 
 **!python tools/train.py --config anomalib/models/padim/config.yaml I got this error :**

Traceback (most recent call last):
File "tools/train.py", line 19, in
from anomalib.data.utils import TestSplitMode
ImportError: cannot import name 'TestSplitMode' from 'anomalib.data.utils' (/usr/local/lib/python3.8/dist-packages/anomalib/data/utils/init.py)

**Reply by samet-akcay** [reply]
@nargolakbari, you need to properly install `anomalib` on google colab as well. The colab version installs anomalib via `pip install anomalib`. The fix @djdameln added today has not been added to pypi yet. Therefore you need to install `anomalib` from source instead. You could refer to the issue #833 and the fix #836 .

**Reply by nargolakbari** [reply]
Thanks for your response , I mounted my Google drive that contains MVTec data set to prevent downloading that  and ran the code .
